### ISSUE TITLE
Ctrl+C is ignored on macOS

### ISSUE THREAD

**Issue by deftomat** [sindresorhus/ora#130]
Change introduced by https://github.com/sindresorhus/ora/pull/112 is ignoring ctrl+c on macOS.

Reverting back to v3 or using `discardStdin: false` fixes it.

Ora version: 4.0.1
macOS version: 10.15 Beta (Catalina)

**Comment by sindresorhus** [comment]
// @stroncium

**Comment by deftomat** [comment]
Looks like it is somehow connected to `execa`. 

Still not sure why it is happening but it looks like it doesn't work when I awaiting `execa` results and running the spinner at the same time. Without `execa`, it works as expected.

But, still, reverting to v3 or using `discardStdin: false` fix this issue.

**Comment by stroncium** [comment]
Could you post some reduced example then(need to see ora options, execa options)? It may be the problem isn't unique to macs.

**Comment by deftomat** [comment]
Sure.

[Here](https://github.com/deftomat/link-with/blob/53762f73db13ed1027a02966b988a850cfe53176/src/link.ts#L79) is spinner and [here](https://github.com/deftomat/link-with/blob/53762f73db13ed1027a02966b988a850cfe53176/src/link.ts#L100) is execa. You can hit ctrl+c during the "Installing transitive dependencies" and it will stop. However, when you update `ora` to v4. It will be ignored. Another spinner defined [here](https://github.com/deftomat/link-with/blob/53762f73db13ed1027a02966b988a850cfe53176/src/link.ts#L54) could be stopped by ctrl+c on both versions.

The package is quite fresh, so maybe it will not work as expected 😉

**Comment by carnesen** [comment]
I've reproduced this issue on macOS and Linux. The special handling of CTRL-C in #112 is not working as intended. I can confirm the workaround of reverting back to v3 or using `discardStdin: false`. This doesn't have anything to do with `execa` since that is neither a dependency of this project nor of mine where I encountered this issue. The issue only manifests on the SECOND spinner, not the first. Here is a simple repro script:
```
const Ora = require('ora');

const firstSpinner = new Ora('First spinner');
firstSpinner.start();

setTimeout(() => {
  firstSpinner.succeed();
  const secondSpinner = new Ora('Second spinner');
  secondSpinner.start();

  setTimeout(() => {
    secondSpinner.succeed();
  }, 3000);
}, 3000);
```

Hit CTRL-C during "First spinner" terminates the program. During "Second spinner" CTRL-C does not terminate the program as expected.
